<?xml version="1.0"?>
<launch>
  <!-- Ideal formation position -->
  <rosparam param="formation_positions">[[0, 0.3], [0, -0.3], [0.6, 0.3], [0.6, -0.3]]</rosparam>

  <!-- Robots numbers -->
  <rosparam param="robots_number">[1, 2, 3, 4]</rosparam>

  <!-- List of neighbours for each robot-->
  <rosparam param="neighbours">[[2, 3], [1, 4], [1, 4], [2, 3]]</rosparam>

  <!-- CBF Communication maintenance, obstacle avoidance, arena, obstacle and extra human activation (1 is activated/0 is deactivated) -->
  <param name="cm" value="1"/>
  <param name="oa" value="1"/>
  <param name="arena" value="1"/>
  <param name="obstacle" value="1"/>
  <param name="extra" value="0"/>

  <!-- Safe distance for communication maintenance, obstacle avoidance, obstacle and extra human -->
  <param name="d_cm" value="0.9"/>
  <param name="d_oa" value="0.3"/>
  <param name="d_obstacle" value="0.35"/>
  <param name="d_extra" value="0.75"/>

  <!-- Variable to determine if HuIL is active or not (1 is activated/0 is deactivated) as well as the robot it affects -->
  <param name="huil" value="1"/>
  <param name="human_robot" value="4"/>

  <!-- Nexus 0  -->
  <group ns="nexus0">
    <!-- Load robot description -->
    <include file="$(find sml_nexus_description)/launch/sml_nexus_description.launch" />
  </group>
    
  <!-- Nexus 1  -->
  <group ns="nexus1">
    <!-- Load robot description -->
    <include file="$(find sml_nexus_description)/launch/sml_nexus_description.launch" />
  </group>

  <!-- Nexus 2  -->
  <group ns="nexus2">
    <!-- Load robot description -->
    <include file="$(find sml_nexus_description)/launch/sml_nexus_description.launch" />
  </group>

  <!-- Nexus 3  -->
  <group ns="nexus3">
    <!-- Load robot description -->
    <include file="$(find sml_nexus_description)/launch/sml_nexus_description.launch" />
  </group>

  <!-- Distributed auto node -->
  <group ns="Auto">
    <node name="k_cbf_huil_dist" pkg="first_task_formation_cbf" type="k_cbf_huil_dist.py" output="screen" >
    </node>
  </group>

  <!-- Human-In-The-Loop controller node -->
  <group ns="HuIL">
    <node name="key_teleop" pkg="first_task_formation_cbf" type="key_teleop.py" output="screen" launch-prefix="gnome-terminal --command" >
      <param name="forward_rate" value="0.3"/>
      <param name="backward_rate" value="0.3"/>
      <param name="rotation_rate" value="0.3"/>
      <param name="hz" value="50"/>
    </node>
  </group>

  <!-- Logger node -->
  <group ns="Logger">
    <node name="logger" pkg="first_task_formation_cbf" type="logger.py" output="screen" >
      <param name="show_motion" value="0"/>
      <param name="cbf_cm_filename" value="/home/victor/catkin_ws/src/cbf-enabled-hil-control-mas/auxiliary_code/cbf_cm_log.csv"/>
      <param name="cbf_oa_filename" value="/home/victor/catkin_ws/src/cbf-enabled-hil-control-mas/auxiliary_code/cbf_oa_log.csv"/>
      <param name="cbf_arena_top_filename" value="/home/victor/catkin_ws/src/cbf-enabled-hil-control-mas/auxiliary_code/cbf_arena_top_log.csv"/>
      <param name="cbf_arena_right_filename" value="/home/victor/catkin_ws/src/cbf-enabled-hil-control-mas/auxiliary_code/cbf_arena_right_log.csv"/>
      <param name="cbf_arena_bottom_filename" value="/home/victor/catkin_ws/src/cbf-enabled-hil-control-mas/auxiliary_code/cbf_arena_bottom_log.csv"/>
      <param name="cbf_arena_left_filename" value="/home/victor/catkin_ws/src/cbf-enabled-hil-control-mas/auxiliary_code/cbf_arena_left_log.csv"/>
      <param name="cbf_obstacle_filename" value="/home/victor/catkin_ws/src/cbf-enabled-hil-control-mas/auxiliary_code/cbf_obstacle_log.csv"/>
      <param name="controller_filename" value="/home/victor/catkin_ws/src/cbf-enabled-hil-control-mas/auxiliary_code/controller_log.csv"/>
      <param name="nom_controller_filename" value="/home/victor/catkin_ws/src/cbf-enabled-hil-control-mas/auxiliary_code/nom_controller_log.csv"/>
      <param name="huil_controller_filename" value="/home/victor/catkin_ws/src/cbf-enabled-hil-control-mas/auxiliary_code/huil_controller_log.csv"/>
    </node>
  </group>

  <!-- Motion capture system SML -->
  <include file="$(find mocap_qualisys)/launch/qualisys.launch" />
  <!-- Motion capture system simulation -->
  <!-- <include file="$(find mocap_simulator)/launch/qualisys_simulator.launch" /> -->

</launch>
